rust/sbp/src/parser/mod.rs

```diff
@@ -153,61 +153,61 @@ impl Parser {
     }
 }
 
-pub fn read_string(buf: &mut dyn Read) -> Result<String> {
-    let mut s = String::new();
-    buf.read_to_string(&mut s)?;
-    Ok(s)
+pub(crate) fn read_string(buf: &mut dyn Read) -> Result<String> {
+    let mut string_buffer = [0u8; crate::SBP_MAX_PAYLOAD];
+    let len = buf.read(&mut string_buffer)?; // TODO: figure out how to get rid of this copy
```

Comment:
To get rid of the extra copy we could modify the signature of this function to take in a `&mut &[u8]` like the other read functions. I think we would need to then find the null terminator and then pass that subslice into `String::from_utf8_lossy`. And something similar could be done to `read_string_limit()`.

Reply:
I tried `&[u8]` too but had issues with "short" reads during testing, this version preserved existing semantics but got rid of the panic on invalid UTF-8 data.

Reply:
I didn't fully understand what you wrote, but I looked at the implementation of `Read` for `&[u8]` to understand how to use `&mut &[u8]` instead of this copy.